contracts/Strategy0xDAOStaker.sol

```diff
@@ -348,86 +374,64 @@ contract Strategy0xDAOStaker is BaseStrategy {
         }
 
         // send our claimed emissionToken to the new strategy
-        emissionToken.safeTransfer(
-            _newStrategy,
-            emissionToken.balanceOf(address(this))
-        );
+        uint256 emis = emissionToken.balanceOf(address(this));
+        if(emis > 0){
+            emissionToken.safeTransfer(
+                _newStrategy,
+                emis
+            );
+        }
+        
     }
 
     ///@notice Only do this if absolutely necessary; as assets will be withdrawn but rewards won't be claimed.
     function emergencyWithdraw() external onlyEmergencyAuthorized {
         masterchef.emergencyWithdraw(pid);
     }
 
+    function manualSell(uint256 _amount) external onlyEmergencyAuthorized {
+        _sell(_amount);
+    }
+
     // sell from reward token to want
     function _sell(uint256 _amount) internal {
-        // sell our emission token for usdc
+
+        if(maxSell > 0){
+            _amount = Math.min(maxSell, _amount);
+        }
+        
+
+        // sell our emission token for pool two second token
         address[] memory emissionTokenPath = new address[](2);
         emissionTokenPath[0] = address(emissionToken);
-        emissionTokenPath[1] = address(usdc);
+        emissionTokenPath[1] = address(poolTwoSecondToken);
```

Comment:
Might this ever need to be swapped via `weth`/`wftm`? In the case where `address(want) == address(poolTwoSecondToken)`?

Reply:
Nevermind, this should be handled just fine as-is

Reply:
![telegram-cloud-photo-size-4-6037309243520038344-x](https://user-images.githubusercontent.com/88827838/152019381-24a37f92-f521-4964-99e8-deaa8d52c572.jpg)
if want == secondtoken then we dont do the second